Lib/locale.py

```diff
@@ -637,27 +637,27 @@ def resetlocale(category=LC_ALL):
 
 
 try:
-    from _locale import _get_locale_encoding
+    from _locale import getencoding
 except ImportError:
-    def _get_locale_encoding():
+    def getencoding():
         if hasattr(sys, 'getandroidapilevel'):
             # On Android langinfo.h and CODESET are missing, and UTF-8 is
             # always used in mbstowcs() and wcstombs().
             return 'UTF-8'
-        if sys.flags.utf8_mode:
-            return 'UTF-8'
         encoding = getdefaultlocale()[1]
         if encoding is None:
-            # LANG not set, default conservatively to ASCII
-            encoding = 'ascii'
+            # LANG not set, default to UTF-8
+            encoding = 'UTF-8'
```

Comment:
This change looks correct according to getencoding() documentation. I consider that it's the same case than when nl_langinfo(CODESET) returns an empty string.

Reply:
Yes.
https://github.com/python/cpython/blob/6927632492cbad86a250aa006c1847e03b03e70b/Python/fileutils.c#L905-L910